### PR TITLE
Add the Madvillian's mask

### PR DESCRIPTION
**Pull request by c420-o**
## About The Pull Request
This adds a tiny but sentimental Item into the game. The Madvillian's mask, previously owned by DOOM, is now available in the hacked autodrone. It can be used for roleplay purposes, such as a cult leader who goes by the pseudonym of "The Madvillian", while his real identity and motives are unknown. Or an Abductor named King Geedorah, ruler of planet x.

Or maybe he's a shitposting clown who raps over comms about cheetos, doritos and fritos. It's up to you. 

## Why It's Good For The Game
It's a pretty lightweight PR, doesn't change anything, just adds things so it doesn't interfere with anything. There's only one way to get it, and only one available, so it's not like it's gonna be spammed very much. I think it's a nice way to memorialize the late MF DOOM, and to also have fun with an absurd personality in an absurd game

## Changelog
:cl:
add: Added string DOOMspeak
add: Madvillian's mask
/:cl:

### CODE REVIEW

File: code/modules/clothing/masks/miscellaneous.dm [hunk 1]
```diff
@@ -184,3 +184,31 @@
 			message = replacetextEx(message,regex(capitalize(key),"g"), "[capitalize(value)]")
 			message = replacetextEx(message,regex(key,"g"), "[value]")
 	speech_args[SPEECH_MESSAGE] = trim(message)
+
+/obj/item/clothing/mask/Madvilliansmask
```
Comment (Jared-Fogle) [inline code/modules/clothing/masks/miscellaneous.dm:188]: Change this entire type path to `/obj/item/clothing/mask/madvillain`, "mask" is implied by `/mask` and only snake_case names are allowed for type paths.

File: code/modules/clothing/masks/miscellaneous.dm [hunk 2]
```diff
@@ -184,3 +184,31 @@
...
+	name = "the MadVillian's mask"
+	desc = "A metal mask formerly owned by the Madvillian."
+	icon_state = "madvilliansmask"
+	inhand_icon_state = "madvilliansmask"
+	w_class = WEIGHT_CLASS_SMALL
+	modifies_speech = TRUE
+
+/obj/item/clothing/mask/Madvilliansmask/dropped(mob/M)
+	. = ..()
+	UnregisterSignal(M, COMSIG_MOB_SAY)
```
Comment (Jared-Fogle) [inline code/modules/clothing/masks/miscellaneous.dm:198]: I don't understand this, you're not registering any signals.

File: code/modules/clothing/masks/miscellaneous.dm [hunk 3]
```diff
@@ -184,3 +184,31 @@
 			message = replacetextEx(message,regex(capitalize(key),"g"), "[capitalize(value)]")
 			message = replacetextEx(message,regex(key,"g"), "[value]")
 	speech_args[SPEECH_MESSAGE] = trim(message)
+
+/obj/item/clothing/mask/Madvilliansmask
+	name = "the MadVillian's mask"
+	desc = "A metal mask formerly owned by the Madvillian."
```
Comment (Jared-Fogle) [inline code/modules/clothing/masks/miscellaneous.dm:190]: Your casing is inconsistent.

File: code/modules/clothing/masks/miscellaneous.dm [hunk 4]
```diff
@@ -184,3 +184,31 @@
...
+	. = ..()
+	UnregisterSignal(M, COMSIG_MOB_SAY)
+
+/obj/item/clothing/mask/Madvilliansmask/handle_speech(datum/source, list/speech_args)
+	var/message = speech_args[SPEECH_MESSAGE]
+	if(message[1] != "*")
+		message = " [message]"
+		var/list/DOOMspeak_words = strings("DOOMspeak.json", "DOOMspeak")
+		for(var/key in DOOMspeak_words)
+			var/value = DOOMspeak_words[key]
```
Comment (Jared-Fogle) [inline code/modules/clothing/masks/miscellaneous.dm:206]:
```suggestion
		var/list/doomspeak_words = strings("DOOMspeak.json", "DOOMspeak")
		for(var/key in doomspeak_words)
			var/value = doomspeak_words[key]
```

Comment (Jared-Fogle) [reply code/modules/clothing/masks/miscellaneous.dm:206]: It doesn't look like there's DOOMspeak.json anywhere though, this doesn't make sense.

File: code/modules/clothing/masks/miscellaneous.dm [hunk 5]
```diff
@@ -184,3 +184,31 @@
...
+			if(islist(value))
+				value = pick(value)
+			message = replacetextEx(message, " [uppertext(key)]", " [uppertext(value)]")
+			message = replacetextEx(message, " [capitalize(key)]", " [capitalize(value)]")
+			message = replacetextEx(message, " [key]", " [value]")
+		if(prob(2))
+			message += pick(" More cheese than doritos, cheetos, fritos."," Just remember ALL CAPS when you spell the man name")
```
Comment (Jared-Fogle) [inline code/modules/clothing/masks/miscellaneous.dm:213]: These should probably be in your strings file as well.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
This isn't even mergeable in its current state. Can you show video that you've tested this and it works?

File: code/modules/clothing/masks/miscellaneous.dm (hunk 2 above)
Comment (Qustinnus) [inline code/modules/clothing/masks/miscellaneous.dm:198]: why is this here? you never register a signal

File: code/modules/vending/autodrobe.dm [hunk 6]
```diff
@@ -128,7 +128,8 @@
 					  /obj/item/gun/magic/wand/nothing = 2,
 					  /obj/item/clothing/glasses/sunglasses/garb = 2,
 					  /obj/item/clothing/glasses/blindfold = 1,
-					  /obj/item/clothing/mask/muzzle = 2)
+					  /obj/item/clothing/mask/muzzle = 2
```
Comment (Qustinnus) [inline code/modules/vending/autodrobe.dm]: need a comma

File: code/modules/clothing/masks/miscellaneous.dm (hunk 4 above)
Comment (c420-o) [reply code/modules/clothing/masks/miscellaneous.dm:206]: Yeah I forgot that I lifted It from spurdo code which is shitcode

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by Jared-Fogle** [comment]
I also don't like that this is essentially just copied and pasted from `/obj/item/clothing/mask/fakemoustache/italian`, I'd rather their behaviors be condensed down to a component or similar.

File: code/modules/clothing/masks/miscellaneous.dm (hunk 2 above)
Comment (c420-o) [reply code/modules/clothing/masks/miscellaneous.dm:198]: It unregisters the signal COSMIG_MOB_SAY puts out that modifies speech, so when you take it off it no longer modifies words to DOOMSPEAK

**Comment by Qustinnus** [comment]
element is probably better

File: code/modules/clothing/masks/miscellaneous.dm (hunk 4 above)
Comment (c420-o) [reply code/modules/clothing/masks/miscellaneous.dm:206]: Doomspeak is in strings

File: code/modules/clothing/masks/miscellaneous.dm (hunk 2 above)
Comment (c420-o) [reply code/modules/clothing/masks/miscellaneous.dm:198]: UnregisterSignal(M, COMSIG_MOB_SAY) doesn't need to register a signal, It just makes it so when you remove the mask the user stops using DOOMspeak

Comment (Qustinnus) [reply code/modules/clothing/masks/miscellaneous.dm:198]: what? you never register for this signal so why unregister for it.

File: code/modules/clothing/masks/miscellaneous.dm (hunk 4 above)
Comment (Jared-Fogle) [reply code/modules/clothing/masks/miscellaneous.dm:206]: It's not included in your files changed, is this already in the code...?

Comment (c420-o) [reply code/modules/clothing/masks/miscellaneous.dm:206]: It's under commits

Comment (c420-o) [reply code/modules/clothing/masks/miscellaneous.dm:206]: Wait no Nevermind I must've added in in another branch by accident

File: strings/DOOMspeak.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+		"fire": "fire vocaled With ash and molten glass like Eyjafjallajökull",
+		"for": "pour",
+		"gibber": "Meat Grinder",
+		"give it back": "Matter fact, gimme back my bracelet and my Shearling",
+		"hi": "yo, It's Victor",
+		"husk": "little charred body smelted like chicken",
+		"me": "the MADVILLIAN",
+		"murderboning": "dashing in a beat-rhyme crime spree",
+		"my name": "the mad name",
+		"retard": "average bug-a-boo",
```
Comment (Qustinnus) [inline strings/DOOMspeak.json:18]: im not sure if this fits ToS

**Review by c420-o — COMMENTED** [review]
gonna add these chizzes

File: code/modules/clothing/masks/miscellaneous.dm (hunk 2 above)
Comment (Qustinnus) [reply code/modules/clothing/masks/miscellaneous.dm:198]: you dont register a signal you mouthbreather

**Comment by Jared-Fogle** [comment]
...Are you resolving my comments without actually fixing them?

File: code/modules/clothing/masks/miscellaneous.dm [hunk 8]
```diff
@@ -184,3 +184,31 @@
...
+	name = "the Madvillain's mask"
+	desc = "A metal mask formerly owned by the Madvillain."
+	icon_state = "madvilliansmask"
+	inhand_icon_state = "madvilliansmask"
+	w_class = WEIGHT_CLASS_SMALL
+	modifies_speech = TRUE
+
+/obj/item/clothing/mask/Madvillainsmask/dropped(mob/M)
+	. = ..()
+	UnregisterSignal(M, COMSIG_MOB_SAY)
```
Comment (Qustinnus) [inline code/modules/clothing/masks/miscellaneous.dm:198]: this is still an issue

**Comment by c420-o** [comment]
> ...Are you resolving my comments without actually fixing them?

I'll fix them in commits

**Comment by Jared-Fogle** [comment]
Can you leave the comments up until you resolve it? 🥴

**Comment by c420-o** [comment]
> Can you leave the comments up until you resolve it? 🥴

sure

**Review by c420-o — COMMENTED** [review]
unresolved them bc maint

**Review by c420-o — COMMENTED** [review]
comma time

File: code/modules/clothing/masks/miscellaneous.dm [hunk 9]
```diff
@@ -184,3 +184,30 @@
...
+
+/obj/item/clothing/mask/Madvillainsmask
+	name = "the Madvillain's mask"
+	desc = "A metal mask formerly owned by the Madvillain."
+	icon_state = "madvillainsmask"
+	inhand_icon_state = "madvillainsmask"
+	w_class = WEIGHT_CLASS_SMALL
+	modifies_speech = TRUE
+
+/obj/item/clothing/mask/Madvillainsmask/dropped(mob/M)
```
Comment (Qustinnus) [inline code/modules/clothing/masks/miscellaneous.dm:196]: you can just remove these two lines they do nothing

File: code/modules/clothing/masks/miscellaneous.dm [hunk 10]
```diff
@@ -184,3 +184,30 @@
 			message = replacetextEx(message,regex(capitalize(key),"g"), "[capitalize(value)]")
 			message = replacetextEx(message,regex(key,"g"), "[value]")
 	speech_args[SPEECH_MESSAGE] = trim(message)
+
+/obj/item/clothing/mask/Madvillainsmask
```
Comment (Qustinnus) [inline code/modules/clothing/masks/miscellaneous.dm:188]: no capitals in typepaths

**Comment by Jared-Fogle** [comment]
can you please stop web editing

**Comment by c420-o** [comment]
> can you please stop web editing

why

**Comment by Jared-Fogle** [comment]
not only because you need to learn git, but because web editing has a near 100% guarantee that you did not run your code

File: strings/DOOMspeak.json (hunk 7 above)
Comment (AlinaStarkova) [reply strings/DOOMspeak.json:18]: also doesnt make sense, since bug-a-boo means object of fear, not sure what the relation is

**Comment by AlinaStarkova** [comment]
running a test of the code in a local server is ideal, however if you cant do that for whatever reason, the absolute bare minimum you should do is compile your code, takes no effort, just hit a button, and it will tell you any missing commas you have, or well, any glaring issues you have in your code to fix before launching your PR or commit

File: strings/DOOMspeak.json (hunk 7 above)
Comment (c420-o) [reply strings/DOOMspeak.json:18]: bug-a-boo is something that causes more trouble than it's worth

**Comment by Tennessee116** [comment]
> > can you please stop web editing
> 
> why

I ensure you that web-editing is more painful than just pressing:
image

Some people will scream in your ear like a banshee and say download VSC and do this and that, but assuming you figure out how 2 into Git, just using anything that can compile your code over web-editor immediately gives you a 99% increase in success chances.

File: strings/DOOMspeak.json (hunk 7 above)
Comment (AlinaStarkova) [reply strings/DOOMspeak.json:18]: and wouldnt that....... be considered highly offensive? i dont think this line is needed when having it causes offense for no lore related reasons

**Comment by c420-o** [comment]
> Some people will scream in your ear like a banshee and say download VSC and do this and that, but assuming you figure out how 2 into Git, just using anything that can compile your code over web-editor immediately gives you a 99% increase in success chances.

valid

File: strings/DOOMspeak.json (hunk 7 above)
Comment (Jared-Fogle) [reply strings/DOOMspeak.json:18]: it's "retard" floyd is talking about, in case either of you didnt realize

File: code/modules/vending/autodrobe.dm [hunk 11]
```diff
@@ -128,7 +128,8 @@
 					  /obj/item/gun/magic/wand/nothing = 2,
 					  /obj/item/clothing/glasses/sunglasses/garb = 2,
 					  /obj/item/clothing/glasses/blindfold = 1,
-					  /obj/item/clothing/mask/muzzle = 2)
+					  /obj/item/clothing/mask/muzzle = 2,
+					  /obj/item/clothing/mask/Madvillainsmask = 1)
```
Comment (Qustinnus) [inline code/modules/vending/autodrobe.dm:132]: capital still

File: code/modules/clothing/masks/miscellaneous.dm [hunk 12]
```diff
@@ -184,3 +184,27 @@
 			message = replacetextEx(message,regex(capitalize(key),"g"), "[capitalize(value)]")
 			message = replacetextEx(message,regex(key,"g"), "[value]")
 	speech_args[SPEECH_MESSAGE] = trim(message)
+
+/obj/item/clothing/mask/madvillainsmask
+	name = "the Madvillain's mask"
```
Comment (Qustinnus) [inline code/modules/clothing/masks/miscellaneous.dm:189]: did you test this ingame because this wont work like you think itll work

**Comment by Qustinnus** [comment]
image

File: code/modules/clothing/masks/miscellaneous.dm (hunk 12 above)
Comment (c420-o) [reply code/modules/clothing/masks/miscellaneous.dm:189]: I did before the pr. it just says smth like "you throw madvillains mask!"

Comment (Qustinnus) [reply code/modules/clothing/masks/miscellaneous.dm:189]: image

Comment (Qustinnus) [reply code/modules/clothing/masks/miscellaneous.dm:189]: image

Comment (c420-o) [reply code/modules/clothing/masks/miscellaneous.dm:189]: Oh okay I get what you're saying

File: strings/DOOMspeak.json (hunk 7 above)
Comment (c420-o) [reply strings/DOOMspeak.json:18]: > and wouldnt that....... be considered highly offensive? i dont think this line is needed when having it causes offense for no lore related reasons
I wasn't aware ss13 was intended to be politically correct. I guess we gotta put in a pull request to remove the italian mustache and the french beret, too

Comment (Qustinnus) [reply strings/DOOMspeak.json:18]: bro noone cares just follow github's ToS you nonce.

Comment (AlinaStarkova) [reply strings/DOOMspeak.json]: > 
> 
> > and wouldnt that....... be considered highly offensive? i dont think this line is needed when having it causes offense for no lore related reasons
> > I wasn't aware ss13 was intended to be politically correct. I guess we gotta put in a pull request to remove the italian mustache and the french beret, too

point is, bug-a-boo literally doesnt describe a "retard" ... the definition you give of "more trouble than its worth" isnt an issue of political correctness (though we still have to follow githubs ToS), but its also offensive for the sole sake to cause offense, causing offense is fine if it makes sense, this doesnt, its just unnesscary

**Comment by c420-o** [comment]
TL;DR I suck at coding

**Comment by Jared-Fogle** [comment]
his brain was saturated with cocaine and rogaine

**Comment by maxymax13** [comment]
amazing

**Comment by 81Denton** [comment]
RIP MF DOOM
